tests/integration/devfile/cmd_devfile_delete_test.go

```diff
@@ -31,187 +38,250 @@ var _ = Describe("odo devfile delete command tests", func() {
-			helper.CmdShouldPass("odo", "delete", "--project", commonVar.Project, "-f")
 		})
-
-		It("should delete the component created from the devfile and also the owned resources", func() {
-			resourceTypes := []string{"deployments", "pods", "services", "ingress"}
-
-			helper.CmdShouldPass("odo", "create", "nodejs", "--project", commonVar.Project, componentName)
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
-			helper.CmdShouldPass("odo", "url", "create", "example", "--host", "1.2.3.4.nip.io", "--port", "3000", "--ingress")
-
-			if os.Getenv("KUBERNETES") != "true" {
-				helper.CmdShouldPass("odo", "url", "create", "example-1", "--port", "3000")
-				resourceTypes = append(resourceTypes, "routes")
-			}
-
-			helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
-
-			helper.CmdShouldPass("odo", "delete", "--project", commonVar.Project, "-f")
-
-			for _, resourceType := range resourceTypes {
-				commonVar.CliRunner.WaitAndCheckForExistence(resourceType, commonVar.Project, 1)
-			}
+		JustAfterEach(func() {})
+		It("should delete the component", func() {
+			helper.CmdShouldPass("odo", "delete", "-f")
 		})
 
-		It("should delete the component created from the devfile and also the env and odo folders and the odo-index-file.json file with all flag", func() {
-			helper.CmdShouldPass("odo", "create", "nodejs", "--project", commonVar.Project, componentName)
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
-			helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
-
-			helper.CmdShouldPass("odo", "url", "create", "example", "--host", "1.2.3.4.nip.io", "--port", "3000", "--ingress")
-
-			if os.Getenv("KUBERNETES") != "true" {
-				helper.CmdShouldPass("odo", "url", "create", "example-1", "--port", "3000")
-			}
+		When("the component is pushed", func() {
+			JustBeforeEach(func() {
+				helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
+			})
+			JustAfterEach(func() {})
+			It("should delete the component, env, odo folders and odo-index-file.json with --all flag", func() {
+				helper.CmdShouldPass("odo", "delete", "-f", "--all")
 
-			helper.CmdShouldPass("odo", "delete", "--project", commonVar.Project, "-f", "--all")
+				files := helper.ListFilesInDir(commonVar.Context)
+				Expect(files).To(Not(ContainElement(".odo")))
+				Expect(files).To(Not(ContainElement("devfile.yaml")))
+			})
 
-			commonVar.CliRunner.WaitAndCheckForExistence("deployments", commonVar.Project, 1)
+			Context("deleting a component from other component directory", func() {
```

Comment:
Describe seems more appropriate here

Reply:
Is it alright to nest Describe inside a Context block?

Reply:
Yes, for me it's alright. Describes explains what we are testing in a high level, Context explains how we are testing it

https://onsi.github.io/ginkgo/#organizing-specs-with-containers-describe-and-context